frame/ibc-transfer/src/lib.rs

```diff
@@ -0,0 +1,576 @@
+#[cfg(feature = "runtime-benchmarks")]
+mod benchmarking;
+mod weight;
+use frame_support::dispatch::Weight;
+pub use weight::WeightInfo;
+
+use codec::{Decode, Encode};
+use core::{fmt::Formatter, marker::PhantomData};
+use frame_system::ensure_signed;
+use ibc::{
+	applications::transfer::acknowledgement::{
+		Acknowledgement as Ics20Acknowledgement, ACK_ERR_STR,
+	},
+	core::{
+		ics04_channel::{
+			channel::{Counterparty, Order},
+			error::Error as Ics04Error,
+			msgs::acknowledgement::Acknowledgement,
+			packet::Packet,
+			Version,
+		},
+		ics24_host::identifier::{ChannelId, ConnectionId, PortId},
+		ics26_routing::context::{Module, ModuleOutputBuilder, OnRecvPacketAck},
+	},
+	signer::Signer,
+};
+use ibc_trait::{CallbackWeight, IbcTrait};
+pub use pallet::*;
+use scale_info::prelude::{
+	format,
+	string::{String, ToString},
+};
+use sp_std::{prelude::*, str::FromStr};
+
+#[frame_support::pallet]
+pub mod pallet {
+	use super::*;
+	use composable_traits::{
+		currency::CurrencyFactory,
+		defi::DeFiComposableConfig,
+		xcm::assets::{RemoteAssetRegistryInspect, RemoteAssetRegistryMutate},
+	};
+	use frame_support::{
+		dispatch::DispatchResult,
+		pallet_prelude::*,
+		traits::{
+			fungibles::{Inspect, Mutate, Transfer},
+			Get,
+		},
+		PalletId, Twox64Concat,
+	};
+	use frame_system::pallet_prelude::*;
+	use ibc::{
+		applications::transfer::{
+			msgs::transfer::MsgTransfer, Amount, PrefixedCoin, PrefixedDenom, VERSION,
+		},
+		core::ics04_channel::channel::{ChannelEnd, State},
+		signer::Signer,
+	};
+	use ibc_primitives::{runtime_interface, runtime_interface::SS58CodecError};
+	use ibc_trait::{
+		channel_id_from_bytes, connection_id_from_bytes, port_id_from_bytes, IbcTrait,
+		OpenChannelParams,
+	};
+	use primitives::currency::CurrencyId;
+	use sp_runtime::{traits::IdentifyAccount, AccountId32};
+
+	#[derive(
+		frame_support::RuntimeDebug,
+		PartialEq,
+		Eq,
+		scale_info::TypeInfo,
+		Encode,
+		Decode,
+		Copy,
+		Clone,
+		Default,
+		codec::MaxEncodedLen,
+	)]
+	pub struct PalletParams {
+		pub send_enabled: bool,
+		pub receive_enabled: bool,
+	}
+
+	#[derive(
+		frame_support::RuntimeDebug,
+		PartialEq,
+		Eq,
+		scale_info::TypeInfo,
+		Encode,
+		Decode,
+		Clone,
+		Default,
+	)]
+	pub struct TransferParams {
+		/// Valid utf8 string bytes
+		pub to: Vec<u8>,
+		/// Source channel on host chain
+		pub source_channel: Vec<u8>,
+		/// Timestamp at which this packet should timeout in counterparty in nanoseconds
+		pub timeout_timestamp: u64,
+		/// Block height at which this packet should timeout on counterparty
+		pub timeout_height: u64,
+		/// Revision number, only needed when making a transfer to a parachain
+		/// in which case this should be the para id
+		pub revision_number: Option<u64>,
+	}
+
+	/// Configure the pallet by specifying the parameters and types on which it depends.
+	#[pallet::config]
+	pub trait Config: frame_system::Config + balances::Config + DeFiComposableConfig {
```

Comment:
Its a good idea to use `crate_name::Trait` so at a glance,  we can know where its from